### PR TITLE
Transpile output with included depdendencies 

### PR DESCRIPTION
**Pull request by juliushaertl**
Resolves https://github.com/nextcloud/nextcloud-event-bus/issues/145

In order to avoid that apps need to transpile the package themselves rollup is used to output a library bundle which allows to also transpile the semver dependency and include it in the package.

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { DEFAULT_EXTENSIONS } from '@babel/core'
+const extensions = [...DEFAULT_EXTENSIONS, '.ts', '.tsx']
+
+const packageJson = require("./package.json");
+
+export default {
+	input: 'lib/index.ts',
+	output: [
+		{
+			file: 'dist/bundled/index.cjs.js',
```
Comment (ChristophWurst) [inline rollup.config.js:16]: this doesn't match the package.json module path – intended?

Comment (juliushaertl) [reply rollup.config.js:16]: I was thinking about still providing the plain non-transpiled ones, but apparently code-deduplication will only work if the exact versions match (e.g. for core-js/semver) so it is probably not worth the additional complexity. I did some comparison and there is not much saving in space between the babel-only and the bundled packages.

**Comment by juliushaertl** [comment]
Would need a backport to some freshly created stable1.2 branch for older Nextcloud versions that still support IE11 (https://github.com/nextcloud/nextcloud-event-bus/tree/stable1.2-rollup is already tested to work with IE11 with using a webpack config that does exclude all node modules from babel)

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+			file: packageJson.module,
+			format: "esm",
+			sourcemap: true
+		}
+	],
+	plugins: [
+		resolve({ extensions }),
+		typescript(),
+		commonjs({ extensions }),
+		babel({
```
Comment (skjnldsv) [inline rollup.config.js:29]: What babel config is it using?

Comment (juliushaertl) [reply rollup.config.js:29]: The existing one that is already in place from just using babel directly before this pr:
https://github.com/nextcloud/nextcloud-event-bus/blob/master/babel.config.js

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**